plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -125,27 +129,25 @@ object StmtConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext<Re
 
                 val receiver = data.convert(propertyAccessExpression.dispatchReceiver!!)
 
-                return when (val getter = symbol.getter) {
-                    is FirDefaultPropertyGetter -> {
-                        val fieldAccess = Exp.FieldAccess(receiver.toViper(), varEmbedding.toField())
-                        val accPred = AccessPredicate.FieldAccessPredicate(fieldAccess, PermExp.FullPerm())
-
-                        data.withResult(varEmbedding.type) {
-                            // We do not track permissions over time and thus have to inhale and exhale the permission when reading a field.
-                            data.addStatement(Stmt.Inhale(accPred))
-                            data.addStatement(Stmt.assign(resultExp.toViper(), fieldAccess))
-                            resultCtx.resultVar.provenInvariants().forEach {
-                                data.addStatement(Stmt.Inhale(it))
-                            }
-                            data.addStatement(Stmt.Exhale(accPred))
+                return if (symbol.getterSymbol == null || symbol.getter is FirDefaultPropertyGetter) {
```

Comment:
Could you take a look at https://github.com/jesyspa/kotlin/pull/98 and see how these changes would fit with those?

Reply:
(That got merged faster than I expected when I made this comment, I think it should all be fine, though.)

Reply:
I managed to incorporate them